Comet/Server.php

```diff
@@ -2187,11 +2205,14 @@ public function UserWebSessionStart()
 	 * @return \Psr\Http\Message\RequestInterface
 	 */
 	public function AsPSR7(NetworkRequest $nr) {
-
 		$params = $nr->Parameters();
 		$params['Username'] = $this->username;
 		$params['AuthType'] = 'Password';
 		$params['Password'] = $this->password;
+		if((strlen($TOTPCode) > 0) || (strlen($this->TOTPCode) > 0)) {
```

Comment:
Looks like `$TOTPCode` may be undefined here, so this condition could change to be `if(strlen($this->TOTPCode)`